### ISSUE TITLE
Error: failed to create API with "go.kubebuilder.io/v3": exit status 2

### ISSUE THREAD

**Issue by shamsher31** [operator-framework/operator-sdk#4556]
#### What did you do?

Ref: https://sdk.operatorframework.io/docs/building-operators/golang/quickstart/

```
$ mkdir memcached-operator
$ cd memcached-operator/
$ operator-sdk init --domain example.com --repo github.com/example/memcached-operator
```

```
$ operator-sdk create api --group cache --version v1alpha1 --kind Memcached --resource --controller

Error: failed to create API with "go.kubebuilder.io/v3": exit status 2
```

#### What did you expect to see?

`create api` should be successful.

#### What did you see instead? Under which circumstances?

```
$ operator-sdk create api --group cache --version v1alpha1 --kind Memcached --resource --controller

Error: failed to create API with "go.kubebuilder.io/v3": exit status 2
```

#### Environment

**Operator type:**

/language go

**Kubernetes cluster type:**

```
$ kind version
kind v0.10.0 go1.15.7 linux/amd64
```
 
```
$ operator-sdk version
operator-sdk version: "v1.4.2", commit: "4b083393be65589358b3e0416573df04f4ae8d9b", kubernetes version: "v1.19.4", go version: "go1.15.8", GOOS: "linux", GOARCH: "amd64"
```

`$ go version` (if language is Go)
```
$ go version
go version go1.16 linux/amd64
```

`$ kubectl version`
```
$ kubectl version --short
Client Version: v1.19.0
Server Version: v1.20.2
```

**Comment by eglookz** [comment]
I also faced this problem. It turned out to be fixed by downgrading to Go version 1.15.8.

**Comment by camilamacedo86** [comment]
Go version go1.16 is not supported by the tool. 
See that it requires go `1.15`. 

IMO:

- The doc https://sdk.operatorframework.io/docs/building-operators/golang/installation/#additional-prerequisites requires to be properly updated. 
- Also, we could add to https://sdk.operatorframework.io/docs/installation/ that golang requires the specific go version and that is not required for those who will use the tool to build Helm/Ansible-based operators.
